scene/resources/resource_format_text.cpp

```diff
@@ -65,12 +65,8 @@ Error ResourceLoaderText::_parse_sub_resource_dummy(DummyReadData *p_data, Varia
 	String unique_id = token.value;
 
 	if (!p_data->resource_map.has(unique_id)) {
-		Ref<DummyResource> dr;
-		dr.instantiate();
-		dr->set_scene_unique_id(unique_id);
-		p_data->resource_map[unique_id] = dr;
-		uint32_t im_size = p_data->resource_index_map.size();
-		p_data->resource_index_map.insert(dr, im_size);
+		r_err_str = "Found unique_id reference before mapping, sub-resources stored out of order in resource file";
```

Comment:
If I am not mistaken, this was done so external dependencies did not have to be loaded when converting, which would make the export process slower.

Reply:
yes; this doesn't load the resources, but resource file should never have the sub-resources stored out of order. If we reference a unique id before mapping, then we will screw up the indexex of the sub-resources in the binary file.